### ISSUE TITLE
Reduce HLS delay

### ISSUE THREAD

**Issue by gabek** [owncast/owncast#28]
This is a placeholder issue for discussing and researching reducing the ~20seconds of delay from live that I'm currently seeing.

There's a lot of pieces at play with owncast, each adding their own delay into the mix, but I'd love to see this closer to 10 seconds.

**Comment by gabek** [comment]
This session hasn't happened yet, but something worthwhile since low latency HLS is coming out of beta this year:

https://developer.apple.com/videos/play/wwdc2020/10225

**Comment by gabek** [comment]
Reducing the config's `maxNumberInPlaylist` down to something small like `4` makes a positive impact, reducing the delay by about ~5s in my rudimentary tests.

**Comment by gabek** [comment]
It seems like reducing the segment length to something really small like 0.5 seconds helps a bit, but will eventually cause issues as so many more segments have to be created, generating more overhead.

**Comment by gabek** [comment]
From different places I look online it looks like HLS delay of about ~20s is average.  When using low latency HLS it can obviously get much lower.

https://obsproject.com/forum/threads/20-second-stream-delay.81480/

**Comment by SoonyangZhang** [comment]
https://github.com/KwaiVideoTeam/las

**Comment by gabek** [comment]
Hey @SoonyangZhang, thanks for the suggestion!  Right now since we're targeting small servers with requirements to offload video segments to external storage, having something that doesn't use segmented video as a solution is kind of not an option at the current time.

**Comment by gabek** [comment]
For testing I changed the segment length to 1 second and max number of items in playlist to 2 and it worked surprisingly well.  Here's a comparison of original source and the stream:

<img width="310" alt="Screen Shot 2020-07-16 at 2 35 00 PM" src="https://user-images.githubusercontent.com/414923/87725242-c8a96d00-c771-11ea-98c4-4aea8a0c6941.png">

Only a 8 second delay and I saw no side effects in my relatively short test (on a $10/mo server, I doubt anything less would be able to keep up).  Though if something were to happen and buffering began, there's very little room for error there.
